fixieai/client/agent.py

```diff
@@ -0,0 +1,238 @@
+
+if TYPE_CHECKING:
+    import fixieai.client as fixie_client
+
+    FixieClient = fixie_client.FixieClient
+else:
+    FixieClient = Any
+
+
+class Agent:
+    """Provides an interface to the Fixie GraphQL Agent API.
+
+    Args:
+        client: The FixieClient instance to use.
+        handle: The handle of the Agent.
+    """
+
+    def __init__(
+        self,
+        client: FixieClient,
+        handle: str,
+    ):
+        self._client = client
+        self._gqlclient = self._client.gqlclient
+        self._handle = handle
+        self._metadata: Optional[Dict[str, Any]] = None
+        try:
+            self._metadata = self.get_metadata()
+        except ValueError:
+            self._metadata = None
+
+    @property
+    def handle(self) -> Optional[str]:
+        """Return the handle for this Agent."""
+        return self._handle
+
+    @property
+    def agent_id(self) -> Optional[str]:
+        """Return the agentId for this Agent."""
+        if self._metadata is None:
+            return None
+        return self._metadata["agentId"]
+
+    @property
+    def name(self) -> Optional[str]:
+        """Return the name for this Agent."""
+        if self._metadata is None:
+            return None
+        return self._metadata["name"]
+
+    @property
+    def description(self) -> Optional[str]:
+        """Return the description for this Agent."""
+        if self._metadata is None:
+            return None
+        return self._metadata["description"]
+
+    @property
+    def queries(self) -> Optional[List[str]]:
+        """Return the queries for this Agent."""
+        if self._metadata is None:
+            return None
+        return self._metadata["queries"]
+
+    @property
+    def published(self) -> Optional[bool]:
+        """Return the published status for this Agent."""
+        if self._metadata is None:
+            return None
+        return self._metadata["published"]
+
+    def get_metadata(self) -> Dict[str, Any]:
+        """Return metadata about this Agent."""
+
+        query = gql(
+            """
+            query getAgentByHandle($handle: String!) {
+                agentByHandle(handle: $handle) {
+                    agentId
+                    handle
+                    name
+                    description
+                    queries
+                    published
+                    owner {
+                        username
+                    }
+                }
+            }
+        """
+        )
+        result = self._gqlclient.execute(
+            query, variable_values={"handle": self._handle}
+        )
+        if "agentByHandle" not in result or result["agentByHandle"] is None:
+            raise ValueError(f"Cannot fetch agent metadata for {self._handle}")
+        return result["agentByHandle"]
+
+    def create_agent(
+        self,
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+        queries: Optional[List[str]] = None,
```

Comment:
queries should always be pulled from the agent, they should not be written via GQL

Reply:
Ah, OK. I believe right now the GraphQL endpoint allows these to be updated; I will file a bug to remove that capability.